### PR TITLE
Reconcile plugins with production build.ros.org.

### PR DESCRIPTION
**Pull request by nuclearsandwich**
Synchronizes plugins to install with build.ros.org using plugin script.
### Removed plugins since sync:
* [authentication-tokens](https://plugins.jenkins.io/authentication-tokens)
* [credentials-binding](https://plugins.jenkins.io/credentials-binding)
* [docker-commons](https://plugins.jenkins.io/docker-commons)
* [docker-workflow](https://plugins.jenkins.io/docker-workflow)
* [git-server](https://plugins.jenkins.io/git-server)
* [handlebars](https://plugins.jenkins.io/handlebars)
* [momentjs](https://plugins.jenkins.io/momentjs)
* [pipeline-build-step](https://plugins.jenkins.io/pipeline-build-step)
* [pipeline-graph-analysis](https://plugins.jenkins.io/pipeline-graph-analysis)
* [pipeline-input-step](https://plugins.jenkins.io/pipeline-input-step)
* [pipeline-milestone-step](https://plugins.jenkins.io/pipeline-milestone-step)
* [pipeline-model-api](https://plugins.jenkins.io/pipeline-model-api)
* [pipeline-model-declarative-agent](https://plugins.jenkins.io/pipeline-model-declarative-agent)
* [pipeline-model-definition](https://plugins.jenkins.io/pipeline-model-definition)
* [pipeline-model-extensions](https://plugins.jenkins.io/pipeline-model-extensions)
* [pipeline-rest-api](https://plugins.jenkins.io/pipeline-rest-api)
* [pipeline-stage-step](https://plugins.jenkins.io/pipeline-stage-step)
* [pipeline-stage-tags-metadata](https://plugins.jenkins.io/pipeline-stage-tags-metadata)
* [pipeline-stage-view](https://plugins.jenkins.io/pipeline-stage-view)
* [workflow-aggregator](https://plugins.jenkins.io/workflow-aggregator)
* [workflow-basic-steps](https://plugins.jenkins.io/workflow-basic-steps)
* [workflow-cps-global-lib](https://plugins.jenkins.io/workflow-cps-global-lib)
* [workflow-durable-task-step](https://plugins.jenkins.io/workflow-durable-task-step)

### Updated plugins since sync
* [script-security](https://plugins.jenkins.io/script-security) 1.33 -> 1.34
* [token-macro](https://plugins.jenkins.io/token-macro) 2.2 -> 2.3
* [workflow-cps](https://plugins.jenkins.io/workflow-cps) 2.39 -> 2.40

### Downgraded plugins since sync
* [workflow-job](https://plugins.jenkins.io/workflow-job) 2.12.2 -> 2.11.2

### Other changes
* [PrioritySorter](https://plugins.jenkins.io/PrioritySorter) dependency change
* [token-macro](https://plugins.jenkins.io/token-macro) dependency change
* [workflow-job](https://plugins.jenkins.io/workflow-job) dependency change

### CODE REVIEW

**Comment by nuclearsandwich** [comment]
I'm not entirely sure why the removed plugins were removed but as they don't appear to be used in production, we probably won't miss them on new deployments. 

For the updates, script-security is one to pay attention to, but the changes between those versions is fairly minor:
* JENKINS-46391 - Properly handle ~/foo/ regexp declarations and some other Pattern methods.
* JENKINS-46358 - Whitelist a number of StringGroovyMethods including replaceAll, findAll, and more

token-macro moved to the latest version which fixes a conflict with docker in pipeline builds (a problem we don't have since we do not use pipeline)

workflow-cps (Pipeline: Groovy) changes are also minor and don't affect our usage as we don't currently use pipeline jobs.

*    JENKINS-34645 - Serialize array iterators properly
*   JENKINS-46391 - Correctly translate ~/foo/ as a regexp
*  JENKINS-46358 - Support for a number of StringGroovyMethods

The workflow job downgrade is the most confusing. 2.12 is the first version requiring Jenkins 2.60+ and downgrading could have been the result of the earlier plugin version being migrated from the trusty buildfarm by mistake.

The only plugins changed above which appear in ros_buildfarm configuration files are PrioritySorter, which doesn't change version, and script-security, where ros_buildfarm has the updated version resulting from this change.

**Comment by nuclearsandwich** [comment]
On a clean test deploy there is one difference from our current production server which is that script-security no longer depends on bouncycastle-api (though many other plugins still do).

I think it's safe to incorporate this change into the PR since it won't change anything about our production instance and I'm starting a plugin upgrade test run for #193 

```diff
   ::jenkins::plugin { 'script-security':
     version => '1.34',
-    require => [ Jenkins::Plugin['bouncycastle-api'] ]
+    require => [  ]
   }
```

File: modules/profile/manifests/jenkins/rosplugins.pp [hunk 1]
```diff
@@ -442,39 +347,19 @@
...
-
-  ::jenkins::plugin { 'workflow-durable-task-step':
-    version => '2.14',
-    require => [ Jenkins::Plugin['bouncycastle-api'], Jenkins::Plugin['durable-task'], Jenkins::Plugin['script-security'], Jenkins::Plugin['workflow-api'], Jenkins::Plugin['workflow-step-api'], Jenkins::Plugin['workflow-support'] ]
-  }
-
   ::jenkins::plugin { 'workflow-job':
-    version => '2.12.2',
-    require => [ Jenkins::Plugin['workflow-api'], Jenkins::Plugin['workflow-step-api'], Jenkins::Plugin['workflow-support'] ]
+    version => '2.11.2',
```
Comment (tfoote) [inline modules/profile/manifests/jenkins/rosplugins.pp]: Is this an intentional/necessary downgrade?

Comment (nuclearsandwich) [reply modules/profile/manifests/jenkins/rosplugins.pp]: To be perfectly honest, I'm not sure. I haven't figured out the best way to track plugin changes on the production host but the difference between versions doesn't include any described changes that would be issues for us and the downgraded plugin is what's running on the live farm.

This gets us back in sync so I can start testing upgraded plugins across the board so it shouldn't stay downgraded for long.

Comment (tfoote) [reply modules/profile/manifests/jenkins/rosplugins.pp]: +1

**Pull request merged**